cleanlab_studio/studio/enrichment.py

```diff
@@ -0,0 +1,585 @@
+    def id(self) -> str:
+        """
+        (str) ID of the Enrichment Project.
+        """
+        return self._id
+
+    @property
+    def name(self) -> str:
+        """
+        (str) Name of the Enrichment Project.
+        """
+        return self._name
+
+    @property
+    def created_at(self) -> datetime:
+        """
+        (datetime.datetime) When the Enrichment Project was created.
+        """
+        if self._created_at is None:
+            create_at_string = self._get_enrichment_project_dict()["created_at"]
+            self._created_at = _response_timestamp_to_datetime(create_at_string)
+
+        return self._created_at
+
+    @property
+    def updated_at(self) -> datetime:
+        """
+        (datetime.datetime) When the Enrichment Project was last updated.
+        """
+        updated_at = self._get_enrichment_project_dict()["updated_at"]
+        return _response_timestamp_to_datetime(updated_at)
+
+    def to_dict(self) -> Dict[str, Any]:
+        """
+        Returns a dictionary of EnrichmentProject metadata.
+        """
+        return {
+            "id": self.id,
+            "name": self.name,
+            "created_at": self.created_at,
+            "updated_at": self.updated_at,
+        }
+
+    def preview(
+        self,
+        options: EnrichmentOptions,
+        *,
+        new_column_name: str,
+        indices: Optional[List[int]] = None,
+    ) -> EnrichmentPreviewResults:
+        """Enrich a subset of data for a preview."""
+        _validate_enrichment_options(options)
+
+        user_input_regex = options.get("regex")
+        extraction_pattern, replacements = _handle_replacements_and_extraction_pattern(
+            user_input_regex
+        )
+
+        response = api.enrichment_preview(
+            api_key=self._api_key,
+            project_id=self._id,
+            new_column_name=new_column_name,
+            constrain_outputs=options.get("constrain_outputs", None),
+            extraction_pattern=extraction_pattern,
+            indices=indices,
+            optimize_prompt=options.get("optimize_prompt", True),
+            prompt=options["prompt"],
+            quality_preset=options.get("quality_preset", "medium"),
+            replacements=replacements,
+            tlm_options=(
+                cast(Dict[str, Any], options.get("tlm_options"))
+                if options.get("tlm_options")
+                else {}
+            ),
+        )
+        epr = EnrichmentPreviewResults.from_dict(response["results"])
+
+        return epr
+
+    def run(
+        self,
+        options: EnrichmentOptions,
+        *,
+        new_column_name: str,
+    ) -> dict[str, Any]:
+        """Enrich the entire dataset."""
+        _validate_enrichment_options(options)
+
+        user_input_regex = options.get("regex")
+        extraction_pattern, replacements = _handle_replacements_and_extraction_pattern(
+            user_input_regex
+        )
+
+        response = api.enrichment_populate(
```

Comment:
[nit] I would rename enrichment_populate() function to something more akin to run like enrichment_run()